internal/network_client.go

```diff
@@ -0,0 +1,38 @@
+package internal
+
+// NetworkError represents a network operation error with abstracted checking methods.
+type NetworkError interface {
+	error
+	// IsNotFound returns true if the error indicates the requested resource was not found.
+	IsNotFound() bool
+	// IsBadSequence returns true if the error indicates a bad sequence number.
+	IsBadSequence() bool
+	// IsTimeout returns true if the error indicates a timeout occurred.
+	IsTimeout() bool
+	// ResultString returns the result string from the error, if available.
+	ResultString() (string, error)
+}
+
+// NetworkClient defines a general interface for interacting with Stellar network services.
+// It abstracts the functionality needed for friendbot operations, allowing different
+// implementations (Horizon, RPC, etc.) to be used interchangeably.
+type NetworkClient interface {
+	// SubmitTransaction submits a transaction in XDR format and returns the result.
```

Comment:
```suggestion
	// SubmitTransaction, submits a transaction in XDR format and will block until it can return the tx result obtained from network.
```

It looks like the expectation of this method is blocking correct?

Reply:
The fn currently blocks, but RPCs fn won't, and will have to internally poll. So maybe this is the right time to change it to be async, and either add a third fn to the NetworkClient for getting the status, or make friendbot async and not make it friendbot's problem.